### PR TITLE
expression: rewrite builtin function: CONV

### PR DESCRIPTION
**Pull request by winkyao**
for #4080

### CODE REVIEW

File: expression/builtin_math.go [hunk 1]
```diff
@@ -675,71 +675,73 @@ type convFunctionClass struct {
...
+	baseStringBuiltinFunc
 }
 
-// eval evals a builtinConvSig.
-// See https://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_conv
-func (b *builtinConvSig) eval(row []types.Datum) (d types.Datum, err error) {
-	args, err := b.evalArgs(row)
-	if err != nil {
-		return d, errors.Trace(err)
+// evalString evals a builtinConvSig.
```
Comment (zz-jason) [inline expression/builtin_math.go]: how about change it to `evalString evals CONV(N,from_base,to_base).` ?

File: expression/builtin_math.go [hunk 2]
```diff
@@ -751,7 +753,8 @@ func (b *builtinConvSig) eval(row []types.Datum) (d types.Datum, err error) {
 	if negative {
 		val = -val
 	}
-	// See https://github.com/mysql/mysql-server/blob/5.7/strings/longlong2str.c#L58
+
+	// See https://github.com/mysql/mysql-server/blob/5.7/strings/longlong2str.c#L58.
```
Comment (zz-jason) [inline expression/builtin_math.go]: remove the reference to mysql source code

File: expression/builtin_math.go [hunk 3]
```diff
@@ -675,71 +675,73 @@ type convFunctionClass struct {
...
 		n = n[1:]
 	}
 
 	val, err := strconv.ParseUint(n, int(fromBase), 64)
 	if err != nil {
-		return d, types.ErrOverflow.GenByArgs("BIGINT UNSINGED", n)
+		return res, false, types.ErrOverflow.GenByArgs("BIGINT UNSINGED", n)
 	}
-	// See https://github.com/mysql/mysql-server/blob/5.7/strings/ctype-simple.c#L598
+	// See https://github.com/mysql/mysql-server/blob/5.7/strings/ctype-simple.c#L598.
```
Comment (zz-jason) [inline expression/builtin_math.go]: remove the reference to mysql source code

**Comment by winkyao** [comment]
@zz-jason  PTAL

File: expression/integration_test.go [hunk 4]
```diff
@@ -637,6 +637,28 @@ func (s *testIntegrationSuite) TestTimeBuiltin(c *C) {
...
 
+func (s *testIntegrationSuite) TestConvBuiltin(c *C) {
+	defer func() {
+		s.cleanEnv(c)
+		testleak.AfterTest(c)()
+	}()
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+
+	result := tk.MustQuery("SELECT CONV('a', 16, 2);")
```
Comment (XuHuaiyu) [inline expression/integration_test.go]: 1. move this to TestMathBuiltin
2. use one line to test conv is ok,
result = tk.MustQuery("SELECT CONV('a', 16, 2), CONV('6E', 18, 8), ......")
result.Check(testkit.Rows("1010 172 ..."))

Comment (winkyao) [inline expression/integration_test.go]: I think use multiple line is better and is more clear @XuHuaiyu

**Comment by winkyao** [comment]
@XuHuaiyu PTAL

**Pull request merged**